tools/clang/lib/AST/DeclCXX.cpp

```diff
@@ -72,8 +72,9 @@ CXXRecordDecl::DefinitionData::DefinitionData(CXXRecordDecl *D)
       ImplicitCopyAssignmentHasConstParam(true),
       HasDeclaredCopyConstructorWithConstParam(false),
       HasDeclaredCopyAssignmentWithConstParam(false), IsLambda(false),
-      IsParsingBaseSpecifiers(false), HasHLSLLongVector(false), NumBases(0),
-      NumVBases(0), Bases(), VBases(), Definition(D), FirstFriend() {}
+      IsParsingBaseSpecifiers(false), HasHLSLLongVector(false),
+      HasHLSLHitObject(false), NumBases(0), NumVBases(0), Bases(), VBases(),
```

Comment:
I think `HasHLSLHitObject` is overly specific.  I would prefer a flag that maps to our concept of an intangible type that we are using in upstream.  This could be used for any decl that contains any special HLSL intangible object type, such as resource objects, RayQuery, HitObject, and so on.

Reply:
Removed the tracking bit, recursing into the type instead to reach all fields (to identify things like disallowed field types in subobjects of arrays of subobjects of the object that is being declared)